src/pages/index.js

```diff
@@ -81,45 +154,106 @@ const newCardPopup = new PopupWithForm({
 });
 newCardPopup.setEventListeners();
 
+const userInfo = new UserInfo({
+  profileTitleSelector: ".profile__title",
+  profileDescriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+Promise.all([userInfo]);
+api
+  .getUserInfo(userInfo._title, userInfo._description)
+  .then((res) => {
+    userInfo.setUserInfo({ title: res.name, description: res.about });
+    userInfo.setAvatar(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: (data) => {
-    handleProfileFormSubmit(data);
+    profileEditPopup.renderLoading(true);
+    api
+      .editUserInfo({ name: data.title, about: data.description })
+      .then((res) => {
+        userInfo.setUserInfo({ title: res.name, description: res.about });
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        profileEditPopup.renderLoading(false);
+        profileEditPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.